internal/storage/sql/errors.go

```diff
@@ -0,0 +1,98 @@
+package sql
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/lib/pq"
+	"github.com/mattn/go-sqlite3"
+	"go.flipt.io/flipt/internal/storage"
+)
+
+var (
+	errNotUnique          = fmt.Errorf("not unique: %w", storage.ErrInvalid)
+	errForeignKeyNotFound = fmt.Errorf("associated resource: %w", storage.ErrNotFound)
+)
+
+// AdaptError converts specific known-driver errors into wrapped storage errors.
+func (d Driver) AdaptError(err error) error {
```

Comment:
oh i like this. is the idea to replace the specific `sql/x` subpackages that override the `common` implementation with this for existing flag/segment/etc CRUD operations?

Reply:
Yeah, 💯 I think we might be able to do away with having to add adaptors for each driver.
I can port the rest of Flipt to it in a later PR too if we like how it works.